azurecontainerapps/src/ContainerAppHelper.ts

```diff
@@ -17,41 +17,314 @@ export class ContainerAppHelper {
+                );
+            } catch (err) {
+                tl.error(err.message);
+                throw err;
+            }
+    }
+
+    /**
+     * Updates an existing Azure Container App based from a YAML configuration file.
+     * @param containerAppName - the name of the existing Container App
+     * @param resourceGroup - the resource group that the existing Container App is found in
+     * @param yamlConfigPath - the path to the YAML configuration file that the Container App properties will be based from
+     */
+    public updateContainerAppFromYaml(
+        containerAppName: string,
+        resourceGroup: string,
+        yamlConfigPath: string) {
+            tl.debug(`Attempting to update Container App with name "${containerAppName}" in resource group "${resourceGroup}" from provided YAML "${yamlConfigPath}"`);
+            try {
+                let command = `containerapp update -n ${containerAppName} -g ${resourceGroup} --yaml ${yamlConfigPath}`;
+
+                new Utility().throwIfError(
+                    tl.execSync('az', command),
+                    tl.loc('UpdateContainerAppFromYamlFailed')
+                );
+            } catch (err) {
+                tl.error(err.message);
+                throw err;
+            }
+    }
+
+    /**
+     * Determines if the provided Container App exists in the provided resource group.
+     * @param containerAppName - the name of the Container App
+     * @param resourceGroup - the resource group that the Container App is found in
+     * @returns true if the Container App exists, false otherwise
+     */
+    public doesContainerAppExist(containerAppName: string, resourceGroup: string): boolean {
+        tl.debug(`Attempting to determine if Container App with name "${containerAppName}" exists in resource group "${resourceGroup}"`);
+        try {
+            const command = `containerapp show -n ${containerAppName} -g ${resourceGroup} -o none`;
+            const result = tl.execSync('az', command);
+            return result.code == 0;
+        } catch (err) {
+            tl.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App Environment exists in the provided resource group.
+     * @param containerAppEnvironment - the name of the Container App Environment
+     * @param resourceGroup - the resource group that the Container App Environment is found in
+     * @returns true if the Container App Environment exists, false otherwise
+     */
+    public doesContainerAppEnvironmentExist(containerAppEnvironment: string, resourceGroup: string): boolean {
+        tl.debug(`Attempting to determine if Container App Environment with name "${containerAppEnvironment}" exists in resource group "${resourceGroup}"`);
+        try {
+            const command = `containerapp environment show -n ${containerAppEnvironment} -g ${resourceGroup} -o none`;
```

Comment:
Does `environment` work? I noticed `az containerapp env show` is used in:
https://learn.microsoft.com/en-us/cli/azure/containerapp/env?view=azure-cli-latest#az-containerapp-env-show

Reply:
Yep you're so right -- I think this was actually autogenerated by GitHub Copilot 😆 It generated an incorrect submodule a few times when I was creating some of the functions, looks like I forgot to correct this one.